app/questionnaire/dependencies.py

```diff
@@ -0,0 +1,39 @@
+from collections import defaultdict
+
+
+class Dependencies:
+    """represents the dependencies for a group of answers
+    held internally as a dict(string,set) where set contains dependent ids
+    set used to prevent duplicates
+
+    To add a dependency either call add with a single dependency
+    or a+=b where both a and b are Dependencies objects.
+
+    To access dependencies for a question id simply index
+    i.e  a['question_id']
+    indexing is read only
+    """
+    def __init__(self):
```

Comment:
Instead of holding a defaultdict internally maybe consider extending a `defaultdict` for this class.

```
class Dependencies(defaultdict):
  def __init__(self):
    super().__init__(set)

  def update(*args, **kwargs):    
    for other in args:  
      self._merge(other)
    if kwargs:
      self._merge(kwargs)

  def _merge(self, other):
    for key, dependencies in other.answer_dependencies.items():
      self[key] |= dependencies
```

(untested)